### PR TITLE
(feat) Upload images to ECR

### PR DESCRIPTION
**Pull request by tdimnet**
Hi,

this is related to a discussion I had with Kirk. AWS wanted to use our Docker Images but they prefer using them on ECR instead of DockerHub.

I updated all of the Github actions in order to make them push the images to both DockerHub and ECR.
I also updated the release script and created the ECR Repo.

### CODE REVIEW

File: .github/workflows/advertisements-fixed.yml [hunk 1]
```diff
@@ -15,23 +17,43 @@ jobs:
...
       uses: docker/build-push-action@v2
       with:
         context: ./ads-service-fixed
-        platforms: linux/amd64 # TODO: Support linux/arm64?
+        platforms: linux/amd64
         push: true
-        tags: ddtraining/advertisements-fixed:latest
+        tags: |
+          ddtraining/advertisements-fixed:latest
+          public.ecr.aws/b1o7r7e0/ddtraining/ads-service-fixed:latest
```
Comment (arapulido) [inline .github/workflows/advertisements-fixed.yml:58]: why are we changing the name of the image based on the repository? `advertisements-fixed` in dockerhub and `ads-service-fixed` in ECR? we should keep the same name

File: .github/workflows/advertisements.yml [hunk 2]
```diff
@@ -15,23 +17,37 @@ jobs:
...
       uses: docker/build-push-action@v2
       with:
         context: ./ads-service
-        platforms: linux/amd64 # TODO: Support linux/arm64?
+        platforms: linux/amd64
         push: true
-        tags: ddtraining/advertisements:latest
+        tags: |
+          ddtraining/advertisements:latest
+          public.ecr.aws/b1o7r7e0/ddtraining/ads-service:latest
```
Comment (arapulido) [inline .github/workflows/advertisements.yml:52]: same comment about the change of the image name

File: .github/workflows/discounts.yml [hunk 3]
```diff
@@ -15,23 +17,40 @@ jobs:
...
+    - name: Build and push
       uses: docker/build-push-action@v2
       with:
         context: ./discounts-service
-        platforms: linux/amd64 # TODO: Support linux/arm64?
+        platforms: linux/amd64
         push: true
-        tags: ddtraining/discounts:latest
+        tags: | 
+          ddtraining/discounts:latest
```
Comment (arapulido) [inline .github/workflows/discounts.yml:54]: same comment about the different image name

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -25,36 +27,55 @@ jobs:
...
-          docker pull ddtraining/storefront-fixed:latest
-          docker tag ddtraining/storefront-fixed:latest ddtraining/storefront-fixed:$TAG
-          docker push ddtraining/storefront-fixed:$TAG
-          docker pull ddtraining/attackbox:latest
-          docker tag ddtraining/attackbox:latest ddtraining/attackbox:$TAG
-          docker push ddtraining/attackbox:$TAG
+
+          IMAGES=(
+            dtraining/advertisements
+            ddtraining/advertisements-fixed
```
Comment (arapulido) [inline .github/workflows/release.yml:57]: not related to this PR, but we could change it as part of this. We are missing `advertisements-errors` here.

File: .github/workflows/advertisements-fixed.yml (hunk 1 above)
Comment (tdimnet) [reply .github/workflows/advertisements-fixed.yml:58]: Yes, you're right. These are some changes I forgot to update. I worked with other ECR repo during the dev in order to keep the official repo clean.

**Comment by tdimnet** [comment]
Thank you @arapulido for your review.
What did you have in mind for the release section?

By the way, I should also update the doc to let people know about ECR?

**Review by arapulido — CHANGES_REQUESTED** [review]
There is an image name that has changed its name

**Review by arapulido — APPROVED** [review]
LGTM! 🚀

**Review by stanleegoodspeed — COMMENTED** [review]
Hi @tdimnet , what AWS account owns these ECR repositories?

**Comment by tdimnet** [comment]
Hey @stanleegoodspeed!

I send you the name of the account via Slack.

File: .github/workflows/advertisements.yml [hunk 5]
```diff
@@ -8,30 +8,48 @@ on:
...
-      uses: docker/setup-qemu-action@v1
-    - name: Setup Docker Buildx
-      uses: docker/setup-buildx-action@v1
-    - name: Login to Docker Hub
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-east-1
+
+    - name: Login to DockerHub
       uses: docker/login-action@v1
```
Comment (burningion) [inline .github/workflows/advertisements.yml:33]: I think we can get rid of pushing to docker hub

File: .github/workflows/attackbox.yml [hunk 6]
```diff
@@ -8,30 +8,54 @@ on:
...
-    - name: Login to Docker Hub
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-east-1
+
+    - name: Login to DockerHub
```
Comment (burningion) [inline .github/workflows/attackbox.yml:38]: Same comment here, let's get rid of automatic pushes to dockerhub, given our conversation

File: .github/workflows/discounts-fixed.yml [hunk 7]
```diff
@@ -8,30 +8,51 @@ on:
...
-    - name: Login to Docker Hub
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-east-1
+
+    - name: Login to DockerHub
```
Comment (burningion) [inline .github/workflows/discounts-fixed.yml:35]: Dockerhub again

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -25,39 +27,61 @@ jobs:
...
+            ddtraining/advertisements-errors
+            ddtraining/discounts
+            ddtraining/discounts-fixed
+            ddtraining/storefront
+            ddtraining/storefront-fixed
+            ddtraining/storefront-no-instrumentation
+            ddtraining/attackbox
+            ddtraining/nginx
```
Comment (burningion) [inline .github/workflows/release.yml:65]: let's get rid of all of these. we don't want to push here (automatically) anymore

**Review by burningion — CHANGES_REQUESTED** [review]
I comment on some (but not all!) places where the dockerhub pushes were occuring. Let's ensure we're only pushing the the ECR registry with our automated builds.

File: .github/workflows/attackbox.yml (hunk 6 above)
Comment (stanleegoodspeed) [reply .github/workflows/attackbox.yml]: Done. All references to dockerhub have been scrubbed

**Review by burningion — APPROVED** [review]

**Pull request merged**